### PR TITLE
nuttx/arch: Remove GCCVER and add compilation options directly.

### PR DESCRIPTION
**Pull request by cuiziwei1**
## Summary
Remove the judgment of GCCVER and add compilation options directly. If there is a problem that the compiler cannot recognize the compilation options, add it separately.

### CODE REVIEW

File: arch/arm64/src/Toolchain.defs [hunk 1]
```diff
@@ -183,13 +183,7 @@ endif
 # Workaround for GCC-12.2 linker warning
 
 ifeq ($(CONFIG_ARCH_TOOLCHAIN_GNU),y)
-  ifeq ($(GCCVER),)
-    export GCCVER := $(shell $(CC) --version | grep gcc | sed -E "s/.* ([0-9]+\.[0-9]+).*/\1/" | cut -d'.' -f1)
-  endif
-
-  ifeq ($(GCCVER),12)
```
Comment (pkarashchenko) [inline arch/arm64/src/Toolchain.defs:190]: The `eq` here is wrong obviously and should be greater equal, but I think that main intension here was to add new flag that appeared starting from GCC 12.

BTW, you remove the condition, but comment above `# Workaround for GCC-12.2 linker warning` is still there ;)

File: arch/arm/src/common/Toolchain.defs [hunk 2]
```diff
@@ -309,16 +309,10 @@ else
   # Wrong warning array subscript [0] is outside array bounds:
   # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523
 
-  ifeq ($(GCCVER),)
-    export GCCVER := $(shell $(CC) --version | grep gcc | sed -E "s/.* ([0-9]+\.[0-9]+).*/\1/" | cut -d'.' -f1)
-  endif
-
-  ifeq ($(GCCVER),12)
     ARCHOPTIMIZATION += --param=min-pagesize=0
```
Comment (pkarashchenko) [inline arch/arm/src/common/Toolchain.defs:312]: Seems now indentations should be updated as well. Also what about comment above?

**Comment by xiaoxiang781216** [comment]
@cuiziwei1 remove the merge patch from pr

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by lupyuen** [comment]
@cuiziwei1 Please check the comments here: https://github.com/apache/nuttx/issues/13776

**Pull request merged**